codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpData.java

```diff
@@ -99,6 +99,23 @@
     long length();
 
     /**
+     * Returns the current defined length of the HttpData.
+     *
+     * If no Content-Length is provided in the request, this is equivalent to the current size (length()) since the
+     * decoding does not known the final size yet. Therefore the defined length is
+     * always 0 the current length (whatever during decoding or in final state).
+     *
+     * If Content-Length is provided in the request, this is this given defined length.
+     * This value does not changed, whatever during decoding or in the final state.
+     *
+     * This method could be used for instance to know the amount of current amount of bytes transmitted for
```

Comment:
My sentence is not good, agree ;-) However I feel like your proposal is not taken into account that the amount of bytes is not fixed yet (growing). But whatever the sentence is wrong in the sens it concerns the `length()` method, not this one... So my full propsosal:

Returns the current defined length of the HttpData.

If no Content-Length is provided in the request, the defined length is always 0 (whatever during decoding or in final state).

If Content-Length is provided in the request, this is this given defined length. This value does not change, whatever during decoding or in the final state.

This method could be used for instance to know the amount of bytes to be transmitted for one particular HttpData, for example one {@link FileUpload} or any known big {@link Attribute}.